src/backend/marsha/core/models/video.py

```diff
@@ -628,6 +631,17 @@ def update_reminders(self, step):
             self.reminders = [step]
         self.save()
 
+    def get_generate_salted_hmac(self):
+        """Generates key used to send emails."""
+        return generate_salted_hmac(settings.REMINDERS_SECRET_HASH, int(self.pk))
```

Comment:
I don't understand why you are casting a UUID in an integer. Can you explain me ?

Reply:
to make it harder to decode, it's an extra confusion

Reply:
In that case I think creating a dedicated non guessable value is a better solution. Obfuscation is not a good solution, I can guess what you are doing by reading the code of the application, it's public. 
A solution can be to create an other column and when a new entry is made you generate a random string and this random string is used as value in the `generate_salted_hmac` function. It's not guessable at all and never public.
And you can use this value for other process if you want.